src/util/Constants.js

```diff
@@ -14,68 +14,50 @@ exports.WSCodes = {
   4014: 'DISALLOWED_INTENTS',
 };
 
-const AllowedImageFormats = ['webp', 'png', 'jpg', 'jpeg', 'gif'];
+const AllowedImageFormats = ['webp', 'png', 'jpg', 'jpeg'];
 
 const AllowedImageSizes = [16, 32, 56, 64, 96, 128, 256, 300, 512, 600, 1024, 2048, 4096];
 
-function makeImageUrl(root, { format = 'webp', size } = {}) {
+function makeImageUrl(root, { hash, format = 'webp', forceStatic = false, size } = {}) {
   if (!['undefined', 'number'].includes(typeof size)) throw new TypeError('INVALID_TYPE', 'size', 'number');
-  if (format && !AllowedImageFormats.includes(format)) throw new Error('IMAGE_FORMAT', format);
+  if (!AllowedImageFormats.includes(format)) throw new Error('IMAGE_FORMAT', format);
   if (size && !AllowedImageSizes.includes(size)) throw new RangeError('IMAGE_SIZE', size);
-  return `${root}.${format}${size ? `?size=${size}` : ''}`;
+  if (forceStatic) format ??= 'webp';
+  else if (hash?.startsWith('a_')) format = 'gif';
+  return `${root}${hash ? `/${hash}` : ''}.${format}${size ? `?size=${size}` : ''}`;
 }
 
 /**
- * Options for Image URLs.
- * @typedef {StaticImageURLOptions} ImageURLOptions
- * @property {boolean} [dynamic=false] If true, the format will dynamically change to `gif` for animated avatars.
- */
-
-/**
- * Options for static Image URLs.
- * @typedef {Object} StaticImageURLOptions
- * @property {string} [format='webp'] One of `webp`, `png`, `jpg`, `jpeg`.
+ * Options for image URLs.
+ * @typedef {Object} ImageURLOptions
+ * @property {string} [format] One of `webp`, `png`, `jpg`, `jpeg`.
```

Comment:
The default is still webp so that shouldn’t be removed
On that topic though, why don’t we change that to png?

Reply:
I don't believe .png is going to be the default option.

I removed the default option here because the output can be .gif. I'm not really sure how to document that as a default?

Reply:
The advantages of using png as the default are obvious but I guess that’s not a topic for here
I think you should keep the default as webp and, on the forceStatic documentation, mention that, if omitted, that will change the format to gif if possible

Reply:
Should be done as of 911f7ad92d5922f12ccd35415aae901fef9aa106!